system/Database/BaseBuilder.php

```diff
@@ -2171,94 +2276,31 @@ protected function validateUpdate(): bool
     }
 
     /**
-     * Compiles an update string and runs the query
+     * Sets data and calls batchExecute to run queryies
+     *
+     * @param array|object|string|null $set         a dataset or select query
+     * @param array|string|null        $constraints
      *
      * @return false|int|string[] Number of rows affected or FALSE on failure, SQL array when testMode
      *
      * @throws DatabaseException
      */
-    public function updateBatch(?array $set = null, ?string $index = null, int $batchSize = 100)
+    public function updateBatch($set = null, $constraints = null, int $batchSize = 100)
```

Comment:
This is probably a BC break.

Reply:
I don't see how this breaks anything. It accepts a string of a single column just like it did before.

Reply:
I'm worried about the scenario when someone would use named parameters. It's a public method.

Reply:
Well we can keep the name index.. its just seemed out of place. Do you think that would be better? Or should we add this as a breaking change?

Reply:
If you change a type definition of a public/protected method, it is a BC. Extended classes will break.

All BCs must be documented in the changelogs.
See https://codeigniter4.github.io/CodeIgniter4/installation/backward_compatibility_notes.html

Reply:
I prefer a proper name. In this case, we need to change the type definitions, they are already BCs.
So I don't mind changing the variable names along with them.